File: leetcode-solutions/arithmetic-subarrays.js
var checkArithmeticSubarrays = function(nums, l, r) {
    let answer = []; 
    for(let i = 0 ; i < l.length; i++){
        let arithmetic = true;
        let sub1 = nums.slice(l[i],r[i]+1);
        sub1 = sub1.sort((a,b) => a-b);
        let target = sub1[0]-sub1[1];
        for(let i = 1; i < sub1.length ; i++){
            if(sub1[i-1] - sub1[i] === target) 
                continue;
            else{
                arithmetic = false;
                break;
            }
        }
        answer.push(arithmetic);
    }
    return answer;
};

File: leetcode-solutions/arithmetic-subarrays.py
class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        output = []
        for i in range(len(l)):
            new_nums = nums[l[i]:r[i]+1]
            new_nums.sort()
            left = 0
            right= 1      
            
            if len(new_nums) == 1:
                output.append(True)
                continue
            diff = new_nums[1] - new_nums[0]
                
            flag = True
            while right < len(new_nums):
                if new_nums[right] - new_nums[left] != diff:
                    flag = False
                    break
                
                left += 1
                right += 1
            output.append(flag)  
        return output